v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/database/CypherGenerator.java

```diff
@@ -190,19 +191,21 @@ private static String getPropertiesListCypherFragment(
    *     com.google.cloud.teleport.v2.neo4j.model.InputRefactoring}
    * @return a list of Cypher schema statements
    */
-  public static Set<String> getIndexAndConstraintsCypherStatements(Target target) {
+  public static Set<String> getIndexAndConstraintsCypherStatements(
+      Target target, Neo4jCapabilities capabilities) {
     TargetType type = target.getType();
     switch (type) {
       case node:
-        return getNodeIndexAndConstraintsCypherStatements(target);
+        return getNodeIndexAndConstraintsCypherStatements(target, capabilities);
       case edge:
-        return getRelationshipIndexAndConstraintsCypherStatements(target);
+        return getRelationshipIndexAndConstraintsCypherStatements(target, capabilities);
       default:
-        throw new IllegalArgumentException(String.format("unexpected target type: %s", type));
+        return Collections.emptySet();
```

Comment:
is the empty set later handled correctly? Jus curious since the current behavior indicates this is not the error that can be retried.

Reply:
It used to throw before, as this method was only called for targets for which we can generate constraints (node and relationship targets).
This PR changes the call site in a way that this is called for every target so returning an empty set of constraints work well here (e.g.: query targets don't have any associated constraints to initialize).

Reply:
Yes, that's fine. Not only caller handles an empty set correctly, but also there is already an if block that will prevent this line to execute.